model/ItemModel.ts

```diff
@@ -46,6 +55,57 @@ class ItemModel {
             response.json(numberOfLists) ;
         });
     }
+    /**
+     * Retrieve a single item by ID
+     *@param response - response object
+     *@param filter - filter object
+     *@returns - JSON Object
+     */
+   public retrieveItemByID(response: any, filter: Object) {
+       // TODO: Check if this is working
+       console.log("Query single item with filter: " + JSON.stringify(filter));
+       var query = this.model.findOne(filter);
+       query.exec((err, itemArray) => {
+          // response.json(itemArray);
+           return JSON.stringify(itemArray);
+
+    });
+    }
+    /**
+     * Get a single item by ID
+     *@param request - request object
+     *@param response - response object
+     */
+
+    public getItem(request:any,response:any): any {
+        const params= request.params;
+        console.log(params);
+        var query = this.model.findOne(params);
+        query.exec( (err, itemArray) => {
+            response.json(itemArray) ;
```

Comment:
Add an error check in case the query fails to execute

Reply:
done